refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java

```diff
@@ -256,6 +248,19 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
         return null;
     }
 
+    private void changeRefactoringElement(RefactoringElementImplementation element, boolean change) {
+        boolean enabled = element.isEnabled();
+        boolean guarded = element.getStatus() == RefactoringElement.GUARDED;
+        boolean readOnly = element.getStatus() == RefactoringElement.READ_ONLY;
+        if (enabled && !(guarded || readOnly)) {
```

Comment:
Hi Carsten, this computes "enabled", "guarded" and "readOnly" even when "element.isEnabled()" is false. The original expression 
`element.isEnabled() && !((element.getStatus() == RefactoringElement.GUARDED) || (element.getStatus() == RefactoringElement.READ_ONLY))`
does not, though.

Reply:
Right, vieiro. I wanted to improve readability, and it resulted in a performance decrease. Haven't even noticed. My bad. Performance is also affected by the added private function call. Before the change, it was done with a redundant inlining of code. Which isn't to bad, but it is not easy to understand.